tests/integ/scala/test_session_suite.py

```diff
@@ -96,6 +96,12 @@ class P1(NamedTuple):
     assert [field.name for field in df.schema.fields] == ["A", "B", "C"]
 
 
```

Comment:
Can we also have a test case for the second issue described here https://snowflake.slack.com/archives/C02EMSN23HD/p1663287588067719? (also let's move these tests to test_dataframe.py? I usually created `create_dataframe` related tests there)

Reply:
Added tests for other special chars and reusing the column names with double quotes.
Both problems are because of the wrong actions on double quotes.